tests/unit/test_svs.cpp

```diff
@@ -2553,6 +2554,128 @@ TEST(SVSTest, quant_modes) {
+        };
+
+        VecSimParams index_params = CreateParams(params);
+        VecSimIndex *index = VecSimIndex_New(&index_params);
+        if (index == nullptr) {
+            if (std::get<1>(svs_details::isSVSQuantBitsSupported(quant_bits))) {
+                GTEST_FAIL() << "Failed to create SVS index";
+            } else {
+                GTEST_SKIP() << "SVS LVQ is not supported.";
+            }
+        }
+
+        std::vector<std::array<float, dim>> v(n);
+        for (size_t i = 0; i < n; i++) {
+            GenerateVector<float>(v[i].data(), dim, i);
+        }
+
+        std::vector<size_t> ids(n);
+        std::iota(ids.begin(), ids.end(), 0);
+
+        auto svs_index = dynamic_cast<SVSIndexBase *>(index);
+        ASSERT_NE(svs_index, nullptr);
+        svs_index->addVectors(v.data(), ids.data(), n);
+
+        ASSERT_EQ(VecSimIndex_IndexSize(index), n);
+
+        float query[] = {50, 50, 50, 50};
+        auto verify_res = [&](size_t id, double score, size_t idx) {
+            EXPECT_DOUBLE_EQ(VecSimIndex_GetDistanceFrom_Unsafe(index, id, query), score);
+            EXPECT_EQ(id, (idx + 45));
+        };
+        runTopKSearchTest(index, query, k, verify_res, nullptr, BY_ID);
+
+        fs::path tmp{fs::temp_directory_path()};
+        auto subdir = "vecsim_test_" + std::to_string(std::rand());
+        auto index_path = tmp / subdir;
+        while (fs::exists(index_path)) {
+            subdir = "vecsim_test_" + std::to_string(std::rand());
+            index_path = tmp / subdir;
+        }
+        fs::create_directories(index_path);
+        svs_index->saveIndex(index_path.string());
+        VecSimIndex_Free(index);
+
+        // Recreate the index from the saved path
+        index = VecSimIndex_New(&index_params);
+        svs_index = dynamic_cast<SVSIndexBase *>(index);
+        ASSERT_NE(svs_index, nullptr);
+        svs_index->loadIndex(index_path.string());
+        fs::remove_all(index_path); // Cleanup the saved index directory
+
+        // Verify the index was loaded correctly
+        ASSERT_EQ(VecSimIndex_IndexSize(index), n);
+        runTopKSearchTest(index, query, k, verify_res, nullptr, BY_ID);
+
+        // Test checkIntegrity function on the loaded index
+        auto integrity_result = svs_index->checkIntegrity();
+        EXPECT_TRUE(integrity_result.valid_state) << "Loaded index should have valid state";
+        EXPECT_GT(integrity_result.memory_usage, 0) << "Memory usage should be positive";
+        EXPECT_EQ(integrity_result.index_size, n) << "Index size should match expected";
+        EXPECT_EQ(integrity_result.storage_size, n) << "Storage size should match index size";
+        EXPECT_EQ(integrity_result.label_count, n) << "Label count should match expected";
+        EXPECT_GE(integrity_result.capacity, n) << "Capacity should be at least index size";
+        EXPECT_FALSE(integrity_result.is_multi) << "This should be a single index";
```

Comment:
Since we have a diffrent load logic, consider adding a sanity test for multi as well